### PR TITLE
optimize IsValid

### PR DESCRIPTION
**Pull request by Hecate2**
A continuation of a cancelled optimization in the past
https://github.com/neo-project/neo-devpack-dotnet/pull/1214/commits/34926a0f52503f02f7720c99cf82be377670a890#diff-c681a7389a2321587cf8377a222c861e641e5a942a1b6aea8c7e522074d92112
because a new standalone PR was requested

### CODE REVIEW

**Review by shargon — APPROVED** [review]

**Comment by Hecate2** [comment]
GAS updated. A GAS consumption increase in a case because the old codes throw exception, while current codes returns the result.

**Comment by shargon** [comment]
> GAS updated. A GAS consumption increase in a case because the old codes throw exception, while current codes returns the result.

It's better a false

**Comment by Hecate2** [comment]
Approval and merge?

File: src/Neo.SmartContract.Framework/UInt160.cs [hunk 1]
```diff
@@ -29,11 +29,13 @@ public extern bool IsValid
         {
             [OpCode(OpCode.DUP)]
             [OpCode(OpCode.ISTYPE, "0x28")] //ByteString
-            [OpCode(OpCode.SWAP)]
+            [OpCode(OpCode.JMPIF, "06")]  // to SIZE
```
Comment (Jim8y) [inline src/Neo.SmartContract.Framework/UInt160.cs:32]: is this because Null itself has a type thus no need to do extra null check?

**Review by Jim8y — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**